src/engine/Resources/Sound.ts

```diff
@@ -582,6 +607,25 @@ class WebAudioInstance implements IAudio {
 
       return this._playingPromise;
    }
+   
+   public playWithProfile(){
+     if (this._isPaused) {
+       this._resume();
+     } else if (!this._isPlaying) {
+       this._start();
+     }
+     var point1 = 0.8;
+     var point2 = 0.1;
+     var point3 = 0.1;
+     var point4 = 0.4;
+     
+     while (new Date().getTime() < this._startTime + this._buffer.duration*1000) {
+       var cpt = (new Date().getTime() - this._startTime) / (this._buffer.duration*1000) ;
```

Comment:
`new Date()` is really hard to test because it can't be mocked easily. Could we use a different method of counting time perhaps similar to how `ex.Timer` works? It would mean that we'd have to add a bit more plumbing into sound so that it can be "updated" over time maybe by adding a `update(delta: number)` method to sound, this does pose a challenge of wiring it into the `ex.Engine` update cycle without making sounds too painful to use. 

This API also looks promising and is designed for working with Audio time https://developer.mozilla.org/en-US/docs/Web/API/AudioContext/currentTime